tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+import sys
+import os
+from time import time, sleep
+import shutil
+import datetime
+import csv
+import json
+import errno
+import tempfile
+
+import unittest
+from mock import Mock, patch
+import numpy as np
+import pandas as pd
+from tsfresh import extract_features, extract_relevant_features, select_features
+from tsfresh.examples.test_tsfresh_baseline_dataset import download_json_dataset
+
+from tsfresh_feature_names import TSFRESH_FEATURES, TSFRESH_BASELINE_VERSION
+
+python_version = int(sys.version_info[0])
+baseline_dir = os.path.dirname(os.path.realpath(__file__))
+tests_dir = os.path.dirname(baseline_dir)
+tsfresh_dir = os.path.dirname(tests_dir)
+baseline_ts_json_file = 'data.json'
+baseline_ts_json_baseline = '%s/tsfresh-%s.py%s.%s.features.transposed.csv' % (
+    baseline_dir, TSFRESH_BASELINE_VERSION, str(python_version),
+    baseline_ts_json_file)
+t_fname_out_fail = '%s/tsfresh/examples/data/test_tsfresh_baseline_dataset/tsfresh-unknown-version.py%s.data.json.features.transposed.csv' % (tsfresh_dir, str(python_version))
+test_path = tempfile.mkdtemp()
```

Comment:
I will given it a try, although seeing as getting unittest and tempfile.mkdtemp() was an absolute pain, I am somewhat scared by the prospect of changing it :)

I only used tempfile.mkdtemp() as I looked at how tsfresh structures other tests and tried to do it the same.  I just using /tmp/test_tsfresh and managing that, which eluded me to the fact that unittest run without any particular order and adhoc, which was messing with the csv via the 'a' write context.  Adopting the tempfile.mkdtemp() method fixed that, but I was also horrified when I added logging to see exactly what the hell it was doing.  Personally I am not sure this "test" matches what unittest is meant to do a script could do it just as well and is probably better suited, but I figured maybe I could hack it into unittest.  I know it is not pretty.